### DISCUSSION TITLE
[V2] Parcel v2 UMD Output

### DISCUSSION THREAD

**Discussion by junwen-k** [parcel-bundler/parcel#5583] [Q&A]
Greetings, I've looked through the documentation as well as the issues but can't seem to find a relevant one. 

In parcel v1, we can use cli build
```sh
parcel build src/index.js --global UMD_NAME
```

This may seem like a very basic question, but how to output an `UMD` module using Parcel V2 ?
(So that the script can be imported via a `<script>` tag in browser and the module will be attached to the window object)

I've noticed that in the [documentation](https://v2.parceljs.org/getting-started/migration/#--global) that is it still not possible in V2. If that's the case, when will V2 add support for bundling `UMD` module ?

My current workaround is installing separate parcel version as alias and use that to output an `UMD` module, which isn't ideal, obviously.

**Comment by VivekFitkariwala** [comment]
Is UMD build supported now? I want to use Code Bundling Shared Bundles functionality in Parcel 2 for my project. Or is any way I can expose it to a window variable.

**Reply by junwen-k** [reply]
I've been manually attaching the variable into the window object at the moment. Sadly there isn't anyone from the parcel team that reply to this question.

**Reply by VivekFitkariwala** [reply]
I think I will move to Webpack....

**Comment by danmarshall** [comment]
I also use this feature from v1. I'm glad to see in this discussion that there are others who do. But I'm skeptical that this discussion will turn into an action item for the Parcel dev team. Perhaps we should create a Feature Request issue?

**Comment by miguelcaravantes** [comment]
please add support for umd

**Comment by wadouk** [comment]
It's a workaround to attach it to window.
Don't think they will make stuff like this is what UMD wrapper will do.

**Comment by gerardnico** [comment]
There is a `global` option in the [target outputformat](https://parceljs.org/features/targets/#outputformat) but it does not seem to do anything more than output a IIFE, therefore making a wrapper not possible.

For now, in the [entry file](https://parceljs.org/features/targets/#entries), you can make the mapping directly in the `windows` variable.

```javascript
window['myGlobal'] = class {
    static myFunction(prop) {
        ...
    }
}
```

**Reply by TechQuery** [reply]
SWC supports UMD, I can't figure out the UMD problem of Parcel v2: https://swc.rs/docs/configuration/modules#umd

**Comment by jvanderen1** [comment]
You should be able to generate a UMD package via the following property in your `package.json`:
```json
{
  "browser": "<path>/<to>/<package>.umd.js",
}
```

**Comment by jonasdeluna** [comment]
Still nothing? What are some alternatives that are faster than webpack

**Comment by spacecheeserocks** [comment]
Just wanted to reference my comment on the issue related to this discussion: https://github.com/parcel-bundler/parcel/issues/7312#issuecomment-1620593989

In short, Parcel does appear to emit UMD code (at least in particular configurations) but **does not** provide any mechanism to define the name for the Global variable it will make. 

Providing that we can get some consensus on how best to configure the name, we could have a PR for this ready soon!
